tests/test_core.py

```diff
@@ -7043,7 +7043,7 @@ def encode_utf8(data):
         for i in range(len(data)):
           data[i] = encode_utf8(data[i])
         return data
-      elif isinstance(data, unicode):
+      elif isinstance(data, type(u'')):
```

Comment:
This seems strictly less readable to me.

Reply:
Would you prefer:

```python
try:
  unicode
except NameError:
  unicode = str
```

Reply:
Not really.  Just noticing out loud that this compatibility can make the code harder to read.

How would you do this in python3-only code?

Reply:
I would have written `isinstance(data, str)` here. But perhaps a better way is doing `isinstance(data, bytes)` so `bytes` gets returned raw, and everything goes through `str` and then encoded.